main.tf

```diff
@@ -261,6 +261,10 @@ resource "azurerm_linux_virtual_machine" "vm_linux" {
       condition     = var.network_interface_ids != null || var.new_network_interface != null
       error_message = "Either `new_network_interface` or `network_interface_ids` must be provided."
     }
+    precondition { #Public keys can only be added to authorized_keys file for 'admin_username' due to a known issue in Linux provisioning agent.
+      condition     = alltrue([for value in var.admin_ssh_keys : false if value.username != var.admin_username && value.username != null])
```

Comment:
Would the following expression be better?:

```hcl
condition     = alltrue([for value in var.admin_ssh_keys : value.username == var.admin_username || value.username == null])
```

Reply:
Yes, cleaner and simpler.